### ISSUE TITLE
Not possible to set --worker_gpu=0 with existing checkpoints for transformer model?

### ISSUE THREAD

**Issue by ajbouh** [tensorflow/tensor2tensor#215]
I'm use the following options to use an existing model
  --data_dir=./t2t/data \
  --problems=wmt_ende_tokens_32k \
  --model=transformer \
  --hparams_set=transformer_base \
  --output_dir=./t2t/train/base

That failed with:
```
Cannot assign a device for operation 'save/ShardedFilename_1': Operation was explicitly assigned to /device:GPU:0 but available devices are [ /job:localhost/replica:0/task:0/cpu:0 ]. Make sure the device specification refers to a valid device.
	 [[Node: save/ShardedFilename_1 = ShardedFilename[_device="/device:GPU:0"](save/StringJoin, save/ShardedFilename_1/shard, save/num_shards)]]
```
So I added this option:
  --worker_gpu=0

Now it fails with:
```
  ...
  File ".../venv/lib/python3.5/site-packages/tensor2tensor/utils/expert_utils.py", line 211, in __init__
    assert device_names_or_functions
AssertionError
```

Is a GPU required for all uses of the transformer Tensor2Tensor model?

**Comment by lukaszkaiser** [comment]
It used to work on CPU with --worker_gpu=1 and should do that, so this is a bug, I'm investigating.

**Comment by lukaszkaiser** [comment]
Could you try this for now: `--worker_gpu=0 --locally_shard_to_cpu` as a workaround? Does it work?

**Comment by ajbouh** [comment]
Yes, `--locally_shard_to_cpu` got me past that error message. Haven't tested the resulting SavedModel yet, but this is promising.

Thank you!

**Comment by lukaszkaiser** [comment]
This should be corrected in 1.1.7, just `--worker_gpu=0` should work. Thanks for pointing it out, and please reopen if you see the problem again!
